core/src/main/java/org/springframework/security/authentication/ReactiveAuthenticationManagerResolver.java

```diff
@@ -0,0 +1,8 @@
+package org.springframework.security.authentication;
+
+import org.springframework.security.authentication.ReactiveAuthenticationManager;
+
+@FunctionalInterface
+public interface ReactiveAuthenticationManagerResolver<C> {
+	ReactiveAuthenticationManager resolve(C context);
```

Comment:
That make sense. I initially had `Mono<..>` but, then changed it probably because a part of code the code failed for some reason but, now that we are reducing the change to `Resource Server` only this should be fine 👍